connector/datadogconnector/connector.go

```diff
@@ -67,15 +67,15 @@ func newConnector(set component.TelemetrySettings, _ component.Config, metricsCo
 		in:              in,
 		metricsConsumer: metricsConsumer,
 		tracesConsumer:  tracesConsumer,
-		exit:            make(chan struct{}),
 	}, nil
 }
 
 // Start implements the component.Component interface.
 func (c *connectorImp) Start(_ context.Context, _ component.Host) error {
 	c.logger.Info("Starting datadogconnector")
 	c.agent.Start()
-	if c.metricsConsumer != nil {
+	c.exit = make(chan struct{})
+	if c.metricsConsumer != nil || c.tracesConsumer != nil {
```

Comment:
It is intentional that we only do `go c.run` for only `metricsConsumer`. We don't want to start it in traces pipelines. Please revert.

Reply:
alright - I'll open a separate PR with a test reproducing the problem and file an issue.

Reply:
#30487 is filed